lapis/src/main/kotlin/org/genspectrum/lapis/response/CsvWriter.kt

```diff
@@ -68,8 +76,15 @@ class IanaTsvWriter {
         if (includeHeaders) {
             writeRow(appendable, data.getHeader(), delimiter)
         }
-        data.getCsvRecords().forEach { record ->
-            writeRow(appendable, record.map { it.orEmpty() }, delimiter)
+        data.getCsvRecords().use { csvRecordStream ->
+            try {
+                csvRecordStream.forEach { csvRecord ->
+                    writeRow(appendable, csvRecord.map { it.orEmpty() }, delimiter)
+                }
+            } catch (e: Exception) {
+                log.error(e) { "Error writing Iana CSV/TSV data" }
```

Comment:
I wonder whether we really want to log at error here. This usually indicates that something is wrong with the code and that we should look at it.

But actually this is a case that simply happens because a client can cause it deliberately (same as sending a malformed request). I think debug (or maybe info) should be enough?

Reply:
I wrote a todo for this - if the exception is an IOException this is most likely a client disconnected and it should be logged differently otherwise I don't think we expect an error and should log the error in full